### PR TITLE
Add only filter support and init package removal by package name

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This adds `only` filter support to `zarf package remove` as well as allowing removal of init packages by their `.tar.zst`

## Related Issue

Fixes: #1859
Fixes: #1839

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

**Comment by Racer159** [comment]
Added `needs-tests` since this should have a final init package remove step at the very end now that that is possible.

File: src/pkg/packager/components.go [hunk 1]
```diff
@@ -251,3 +251,17 @@ func (p *Packager) appendIfNotExists(slice []string, item string) []string {
 	}
 	return append(slice, item)
 }
+
+func (p *Packager) requiresCluster(component types.ZarfComponent) (confirmComponent bool) {
```
Comment (YrrepNoj) [inline src/pkg/packager/components.go:255]: nit; it's a bit weird to have a named return but not use it.

**Review by YrrepNoj — APPROVED** [review]
LGTM. Left a nit comment up to you.

File: src/pkg/packager/components.go (hunk 1 above)
Comment (Racer159) [reply src/pkg/packager/components.go]: Good find!  Fixed

**Review by YrrepNoj — APPROVED** [review]

**Pull request merged**